ERCS/erc-x.md

```diff
@@ -0,0 +1,324 @@
+
+PermaLink-ABTs consolidate asset value by allowing multiple subordinate tokens to be linked to a single binding token, providing enhanced composability and reducing fragmentation. By requiring only the binding token to be transferred, all associated assets move in sync, making it easier to manage portfolios and move groups of assets together. This approach fosters collaboration, value accrual, and compatibility across ecosystems, whether for digital identities, RWAs, NFTs, or other on-chain assets.
+
+
+## Specification
+
+### `IABT` (Token Interface)
+
+**NOTES**:
+
+- The following specifications use syntax from Solidity `0.8.27` (or above)
+- Callers MUST handle `false` from `returns (bool success)`. Callers MUST NOT assume that `false` is never returned!
+
+```solidity
+interface IABT {
+    event AssetBoundContractSet(address assetBoundContract);
+    event TokenRevealed(uint256 tokenId);
+
+    function ownerOf(uint256 tokenId) external view returns (address);
+    function tokenExists(uint256 tokenId) external view returns (bool);
+    function totalSupply() external view returns (uint256);
+    function balanceOf(address owner) external view returns (uint256);
+
+    function reveal(uint256[] calldata tokenIds) external payable;
+}
+```
+
+### **Interface events**
+
+#### `AssetBoundContractSet` event
+
+Emitted when the contract is deployed and bound to `assetBoundContract`
+
+```solidity
+event AssetBoundContractSet(address assetBoundContract);
+```
+
+#### `TokenRevealed` event
+
+Emitted when the `tokenId` is revealed
+
+```solidity
+event TokenRevealed(uint256 tokenId);
+```
+
+### **Interface functions**
+
+The functions detailed below MUST be implemented.
+
+#### `ownerOf` function
+
+Returns the owner of the NFT specified by the `tokenId`. Will read from the `assetBoundContract` the owner and return it.
+
+```solidity
+function ownerOf(uint256 tokenId) external view returns (address);
+```
+
+#### `tokenExists` function
+
+Returns true if the owner of the token is not the zero address (`0x0`)
+
+```solidity
+function tokenExists(uint256 tokenId) external view returns (bool);
+```
+
+#### `totalSupply` function
+
+Gets the total amount of tokens stored by the assetBoundContract
+
+```solidity
+function totalSupply() external view returns (uint256);
+```
+
+#### `balanceOf` function
+
+Returns the number of NFTs in the assetBoundContract that an owner has.
+
+```solidity
+function balanceOf(address owner) external view returns (uint256);
+```
+
+### `reveal` function
+
+An optional `reveal` function MAY be implemented to allow pre-allocated tokens to be activated on demand.  
+This method reduces gas consumption compared to traditional minting and simplifies token activation mechanics.
+
+```solidity
+function reveal(uint256[] calldata tokenIds) external payable;
+```
```

Comment:
It might make sense to define this in a separate interface. This approach has the benefit of being more compatible with ERC-165's `supportsInterface` introspection.